### PR TITLE
Better path handling and opt matching

### PR DESCRIPTION
**Pull request by gsfr**

### CODE REVIEW

**Comment by gsfr** [comment]
Sorry, this PR is quickly turning into a catch-all bucket.

File: reaper/dicom_reaper.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def instrument_query(self):
         if scu_series is None:
             return None
         for series in scu_series:
-            if series['NumberOfSeriesRelatedInstances'] is None:
+            if not series['NumberOfSeriesRelatedInstances']:
```
Comment (gsfr) [inline reaper/dicom_reaper.py:37]: This will also catch if `NumberOfSeriesRelatedInstances == 0`, which is probably ok.

**Comment by gsfr** [comment]
@ryansanford Are we good to go here?

**Comment by ryansanford** [comment]
@gsfr `--exact-opt-match` is still working like a substring match for me. I'll push a change that anchors the regex with `^` and `$` which is making the match be exact.

**Comment by ryansanford** [comment]
At this point, this PR LGTM

**Comment by gsfr** [comment]
Great.

For the record, `re.match` anchors the regex to the beginning of the string, but not to the end. The latest change makes this much more explicit.

**Pull request merged**